roles/dnsmasq/templates/cifmw-dnsmasq.conf.j2

```diff
@@ -10,3 +10,4 @@ dhcp-leasefile=/var/lib/dnsmasq/cifmw-dnsmasq.leases
 {% endif %}
 
 conf-dir={{ cifmw_dnsmasq_basedir }},*.conf
+dhcp-hostsdir="{{ cifmw_dnsmasq_basedir }}/dhcp-hosts.d"
```

Comment:
we can trigger the reload whenever we want/need using `ansible.builtin.meta: flush_handlers`
This is what's done in my branches.

Reply:
I am aware of that. But that is something that will be easy for consumers to forget to do. And I think it will make up for some hard-to-diagnose issues. For example if you add a dhcp-host in your play, then start the VM and expect it to get an address and try to perform operations against it - the play fails but soon after the play fails and the handler ran the VM become accessible - to the user that is not going to be easy to understand without a black belt in Ansible.

I actually think we should add the `flush_handler` thing at the end of all the `manage` task files. To enable adding multiple networks and do one restart we can change the "manage_network" interface to use a list like the `manage_host_record.yml`, `manage_forwarder.yml` etc does. But I don't want to do that now - because changing the interface will break your patches.

I am adding reload functionality in https://github.com/openstack-k8s-operators/ci-framework/pull/1801. Essentially "manage_host" can notify the "Reload" handler, while "manage_network", "manage_host_record", "manage_address" etc notify the "Restart" handler.